File: app/src/main/java/com/icure/kses/modoo/utility/PrefManager.java
package com.icure.kses.modoo.utility;

import android.content.Context;
import android.content.SharedPreferences;

/**
 * Created by whan 2020/06/08
 */
public class PrefManager {
    SharedPreferences pref;
    SharedPreferences.Editor editor;
    Context _context;

    // shared pref mode
    int PRIVATE_MODE = 0;

    // Shared preferences constants
    private static final String PREF_NAME = "modoopref";

    private static final String IS_FIRST_TIME_LAUNCH = "IS_FIRST_TIME_LAUNCH";
    private static final String FCM_ACCESS_TOKEN = "FCM_ACCESS_TOKEN";


    public PrefManager(Context context) {
        this._context = context;
        pref = _context.getSharedPreferences(PREF_NAME, PRIVATE_MODE);
        editor = pref.edit();
    }

    public void setFirstTimeLaunch(boolean isFirstTime) {
        editor.putBoolean(IS_FIRST_TIME_LAUNCH, isFirstTime);
        editor.commit();
    }

    public boolean isFirstTimeLaunch() {
        return pref.getBoolean(IS_FIRST_TIME_LAUNCH, true);
    }

    public void setFcmAccessToken(String accessToken){
        editor.putString(FCM_ACCESS_TOKEN, accessToken);
        editor.commit();
    }

    public String getFcmAccessToken(){
        return pref.getString(FCM_ACCESS_TOKEN, null);
    }
}

File: app/src/main/java/com/icure/kses/modoo/utility/PrefManager.kt
package com.icure.kses.modoo.utility

import android.content.Context
import android.content.SharedPreferences

/**
 * Created by whan 2020/06/08
 */
class PrefManager(val _context: Context) {
    var pref: SharedPreferences
    var editor: SharedPreferences.Editor

    // shared pref mode
    var PRIVATE_MODE = 0

    var isFirstTimeLaunch: Boolean
        get() = pref.getBoolean(PREFKEY_IS_FIRST_TIME_LAUNCH, true)
        set(isFirstTime) {
            editor.putBoolean(PREFKEY_IS_FIRST_TIME_LAUNCH, isFirstTime)
            editor.commit()
        }

    var fcmAccessToken: String?
        get() = pref.getString(PREFKEY_FCM_ACCESS_TOKEN, null)
        set(accessToken) {
            editor.putString(PREFKEY_FCM_ACCESS_TOKEN, accessToken)
            editor.commit()
        }

    var suggestionWords: String?
        get() = pref.getString(PREFKEY_SUGGESTION_WORDS, "")
        set(words) {
            editor.putString(PREFKEY_SUGGESTION_WORDS, words)
            editor.commit()
        }

    companion object {
        // Shared preferences constants
        private const val PREF_NAME = "modoopref"
        private const val PREFKEY_IS_FIRST_TIME_LAUNCH = "PREFKEY_IS_FIRST_TIME_LAUNCH"
        private const val PREFKEY_FCM_ACCESS_TOKEN = "PREFKEY_FCM_ACCESS_TOKEN"
        private const val PREFKEY_SUGGESTION_WORDS = "PREFKEY_SUGGESTION_WORDS"
    }

    init {
        pref = _context.getSharedPreferences(PREF_NAME, PRIVATE_MODE)
        editor = pref.edit()
    }
}